component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -561,13 +631,188 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
+            log.error(newAccessResponseCode + ": Authentication Failure. Please apply valid Consumer key, " +
+                    "Consumer secret for generate new access token. ");
+        } else {
+            log.error(newAccessResponseCode + ": Can not obtain new access token.");
+        }
+
+    }
+
+    public ArrayList<String> getAccessToken(DynamicOptions dynamicOptions, String encodedRefreshAuth) {
+        HttpsClient httpsClient = new HttpsClient();
+        ArrayList<String> newAccessTokenArray;
+        if (!HttpConstants.EMPTY_STRING.equals(oauthUsername) &&
+                !HttpConstants.EMPTY_STRING.equals(oauthUserPassword)) {
+            newAccessTokenArray = httpsClient.getPasswordGrandAccessToken(publisherURL, clientStoreFile,
+                    clientStorePass, oauthUsername, oauthUserPassword, encodedRefreshAuth);
+        } else if (!HttpConstants.EMPTY_STRING.equals(refreshToken.getValue(dynamicOptions)) ||
+                !availableRefreshToken.equals(HttpConstants.EMPTY_STRING)) {
+            newAccessTokenArray = httpsClient.getRefreshGrandAccessToken(publisherURL, clientStoreFile,
+                    clientStorePass, encodedRefreshAuth, refreshToken.getValue(dynamicOptions));
+        } else {
+            newAccessTokenArray = httpsClient.getClientGrandAccessToken(publisherURL, clientStoreFile,
+                    clientStorePass, encodedRefreshAuth);
+        }
+        return newAccessTokenArray;
+    }
+
+    public void checkAccessToken(String encodedRefreshAuth, DynamicOptions dynamicOptions,
+                                 List<Header> headersList) {
+        //check the availability of the authorization
+        boolean authAvailability = false;
+        for (Header header : headersList) {
+            if (header.getName().equals(HttpConstants.AUTHORIZATION_HEADER)) {
+                authAvailability = true;
+            }
+        }
+        if (!authAvailability) {
+            //generate encoded base64 auth for getting refresh token
+            if (!accesstokenAvalable) {
+                ArrayList<String> accessTokenArray = getAccessToken(dynamicOptions, encodedRefreshAuth);
+                int responseCode = Integer.parseInt(accessTokenArray.get(0));
+
+                if (responseCode == HttpConstants.SUCCESS_CODE) {
+                    accessToken = "Bearer " + accessTokenArray.get(1);
+                    availableRefreshToken = accessTokenArray.get(2);
+                    accesstokenAvalable = true;
+                    accessTokenCache.setRefreshtoken(encodedRefreshAuth, availableRefreshToken);
+                    accessTokenCache.setAccessToken(encodedRefreshAuth, accessToken);
+                } else if (responseCode == HttpConstants.AUTHENTICATION_FAIL_CODE) {
+                    accessToken = HttpConstants.EMPTY_STRING;
+                    log.error(responseCode + ": Provide valid Consumer key, " +
```

Comment:
If we did not get the token, we cannot continue publishing the message right? We should throw a proper runtime exception here. I believe this was mentioned in the code review as well.

For the error message, how about something like:
```
"Failed to obtain Access Token with HTTP code: " + responseCode + ". Please provide a valid Consumer key and a Consumer secret."
```